code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm

```diff
@@ -214,3 +217,101 @@
+			if(mob_species == /datum/species/plasmaman)
+				uniform = /obj/item/clothing/under/plasmaman
+				head = /obj/item/clothing/head/helmet/space/plasmaman
+				belt = /obj/item/weapon/tank/internals/plasmaman/belt
+			else
+				uniform = /obj/item/clothing/under/rank/miner/lavaland
+			if(mob_species != /datum/species/lizard)
+				shoes = /obj/item/clothing/shoes/workboots/mining
+			gloves = /obj/item/clothing/gloves/color/black
+			mask = /obj/item/clothing/mask/gas/explorer
+			if(prob(20))
+				suit = pickweight(list(/obj/item/clothing/suit/hooded/explorer = 18, /obj/item/clothing/suit/hooded/cloak/goliath = 2))
+			if (mob_species != /datum/species/plasmaman && prob(4))
+				belt = pickweight(list(/obj/item/weapon/storage/belt/mining = 2, /obj/item/weapon/storage/belt/mining/alt = 2))
+			else if (mob_species != /datum/species/plasmaman)
+				belt = /obj/item/weapon/tank/internals/emergency_oxygen/engi
+			if(prob(30))
+				r_pocket = pickweight(list(/obj/item/stack/marker_beacon = 20, /obj/item/stack/spacecash/c1000 = 7,/obj/item/weapon/reagent_containers/hypospray/medipen/survival = 2, /obj/item/borg/upgrade/modkit/damage = 1 ))
+			if(prob(10))
+				l_pocket = pickweight(list(/obj/item/stack/spacecash/c1000 = 7,/obj/item/weapon/reagent_containers/hypospray/medipen/survival = 2, /obj/item/borg/upgrade/modkit/cooldown = 1 ))
+		if("Ashwalker")
+			mob_species = /datum/species/lizard/ashwalker
+			uniform = /obj/item/clothing/under/gladiator/ash_walker
+			if(prob(95))
+				head = /obj/item/clothing/head/helmet/gladiator
+			else
+				head = /obj/item/clothing/head/helmet/skull
+				suit = /obj/item/clothing/suit/armor/bone
+				gloves = /obj/item/clothing/gloves/bracer
+			if(prob(5))
+				back = pickweight(list(/obj/item/weapon/twohanded/bonespear = 3,/obj/item/weapon/twohanded/fireaxe/boneaxe = 2))
+			if(prob(10))
+				belt = /obj/item/weapon/storage/belt/mining/primitive
+			if(prob(30))
+				r_pocket = /obj/item/weapon/kitchen/knife/combat/bone
+			if(prob(30))
+				l_pocket = /obj/item/weapon/kitchen/knife/combat/bone
+		if("Clown")
+			name = pick(GLOB.clown_names)
+			outfit = /datum/outfit/job/clown
+			outfit.backpack_contents = list()
+			if(prob(70))
+				outfit.backpack_contents += pick(list(/obj/item/weapon/stamp/clown = 1, /obj/item/weapon/reagent_containers/spray/waterflower = 1,/obj/item/weapon/reagent_containers/food/snacks/grown/banana = 1, /obj/item/device/megaphone/clown = 1,/obj/item/weapon/reagent_containers/food/drinks/soda_cans/canned_laughter = 1,/obj/item/weapon/pneumatic_cannon/pie = 1))
+			if(prob(30))
+				outfit.backpack_contents += list(/obj/item/stack/sheet/mineral/bananium = pickweight(list( 1 = 3, 2 = 2, 3 = 1)))
+			if(prob(10))
+				l_pocket = /obj/item/weapon/bikehorn/golden
+			if(prob(10))
+				r_pocket = /obj/item/weapon/implanter/sad_trombone
+		if("Golem")
+			mob_species = /datum/species/golem/random
```

Comment:
It depends on whether the species is applied before or after the new human is killed. But yes, none of those golems should be legion-able.